Sources/Sentry/SentryDevice.mm

```diff
@@ -0,0 +1,216 @@
+ * @endcode
+ * @seealso See
+ * https://www.cocoawithlove.com/blog/2016/03/08/swift-wrapper-for-sysctl.html#looking-for-the-source
+ * for more info.
+ * @return @c sysctl value for the combination of @c CTL_HW and the provided other flag in the
+ * type parameter.
+ */
+NSString *
+getHardwareDescription(int type)
+{
+#if SENTRY_HAS_UIKIT && !TARGET_OS_SIMULATOR
+    NSCAssert(type != HW_MODEL, @"Don't call this method with HW_MODEL for (non-simulator) iOS devices");
+#endif
+    int mib[2];
+    char name[128];
+    size_t len;
+
+    mib[0] = CTL_HW;
+    mib[1] = type;
+    len = sizeof(name);
+    if (SENTRY_LOG_ERRNO(sysctl(mib, 2, &name, &len, NULL, 0)) != 0) {
+        return @"";
+    }
+    return [NSString stringWithUTF8String:name];
+}
+
+/**
+ * Provided as a fallback in case @c sysctlbyname fails in @c getCPUArchitecture using the @c hw.cpusubtype option.
+ * @note I've not observed a device that has needed this (armcknight 22 Sep 2022). Tested on:
+ * @code
+ *   - 2015 MBP (x86_64H)
+ *   - 2020 iMac (x86_64H)
+ *   - 2021 MBP (M1 reported as arm64e)
+ *   - iPhone simulators on all of those macs
+ *   - iPhone 13 mini (arm64e)
+ *   - iPod Touch (6th gen) (armv8)
+ * @endcode
+ */
+NSString *
+getCPUType(NSNumber *_Nullable subtype)
+{
+    cpu_type_t type;
+    size_t typeSize = sizeof(type);
+    if (SENTRY_LOG_ERRNO(sysctlbyname("hw.cputype", &type, &typeSize, NULL, 0)) != 0) {
+        if (subtype != nil) {
+            return
+                [NSString stringWithFormat:@"no CPU type for unknown subtype %d", subtype.intValue];
+        }
+        return @"no CPU type or subtype";
+    }
+    switch (type) {
+    default:
+        if (subtype != nil) {
+            return [NSMutableString
+                stringWithFormat:@"unknown CPU type (%d) and subtype (%d)", type, subtype.intValue];
+        }
+        return [NSMutableString stringWithFormat:@"unknown CPU type (%d)", type];
+    case CPU_TYPE_X86_64:
+        // I haven't observed this branch being taken for 64-bit x86 architectures. Rather, the
+        // x86 branch is taken, and then the subtype is reported as the 64-bit
+        // subtype. Tested on a 2020 Intel-based iMac and 2015 MBP. (armcknight 21 Sep 2022)
+        return @"x86_64";
+    case CPU_TYPE_X86:
+        return @"x86";
+    case CPU_TYPE_ARM:
+        return @"arm";
+    case CPU_TYPE_ARM64:
+        return @"arm64";
+    case CPU_TYPE_ARM64_32:
+        return @"arm64_32";
+    }
+}
+} // namespace
+
+NSString *
+getCPUArchitecture(void)
+{
+    cpu_subtype_t subtype;
+    size_t subtypeSize = sizeof(subtype);
+    if (SENTRY_LOG_ERRNO(sysctlbyname("hw.cpusubtype", &subtype, &subtypeSize, NULL, 0)) != 0) {
+        return getCPUType(nil);
+    }
+    switch (subtype) {
+    default:
+        return getCPUType(@(subtype));
+    case CPU_SUBTYPE_X86_64_H:
+        return @"x86_64H";
+    case CPU_SUBTYPE_X86_64_ALL:
+        return @"x86_64";
```

Comment:
All the macs I have enter the branch for `CPU_SUBTYPE_X86_64_H`... do we want to report these as `x86_64H` (or lowercase h) vs just `x86_64`?

Reply:
I think there is a difference between `x86_64` and `x86_64h`, so I think we should report it as `x86_64h`.